### PR TITLE
fix(aad,simpleauth): fix format of telemetry

### PR DESCRIPTION
**Pull request by KennethBWSong**
fix:
1. remove plugin name from event name
2. fix error telemetry:
image

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/simpleauth/constants.ts [hunk 1]
```diff
@@ -94,6 +94,24 @@ export class Constants {
...
+export class Telemetry {
+    static component = "component";
+    static errorCode = "error-code";
+    static errorType = "error-type";
+    static errorMessage = "error-message";
+    static userError = "user";
+    static systemError = "system";
+  }
+  
+  export class Stage {
```
Comment (adashen) [inline packages/fx-core/src/plugins/resource/simpleauth/constants.ts:106]: Are these stage used as event names for error? If so seems not same as the ones defined in https://github.com/OfficeDev/TeamsFx/blob/8f8c5ea16afadb40f21a9c4df32146fd15e37701/packages/fx-core/src/plugins/resource/aad/constants.ts#L95 Besides, please check with Changyong about the names. From the document I didn't see local-debug. However there are f5-start, f5

File: packages/fx-core/package.json [hunk 2]
```diff
@@ -110,7 +111,7 @@
     "faker": "^5.5.2",
     "jwt-decode": "^3.1.2",
     "lint-staged": "^10.5.4",
-    "mocha": "^8.2.1",
+    "mocha": "^8.4.0",
```
Comment (xzf0587) [inline packages/fx-core/package.json:114]: it is needed to upgrade the version?

File: packages/fx-core/tests/plugins/resource/aad/unit/index.test.ts [hunk 3]
```diff
@@ -128,63 +119,6 @@ describe("AadAppForTeamsPlugin: CI", () => {
     const postProvision = await plugin.postLocalDebug(context);
     chai.assert.isTrue(postProvision.isOk());
   });
-
-  it("provision and local debug: tab and bot with update current and wrong permission", async function () {
```
Comment (adashen) [inline packages/fx-core/tests/plugins/resource/aad/unit/index.test.ts:132]: why test case is deleted?

Comment (KennethBWSong) [reply packages/fx-core/tests/plugins/resource/aad/unit/index.test.ts:132]: These test case are for test UpdatePermission Command which are not deleted previously.

File: packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts [hunk 4]
```diff
@@ -15,11 +15,11 @@ export class TelemetryUtils {
     eventName: string,
     properties?: { [key: string]: string },
     measurements?: { [key: string]: number }
-  ) {
+  ): void {
```
Comment (xzf0587) [inline packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts:18]: It is suggested that add a success property value "yes" for successful result and "no" for failed result.

File: packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts [hunk 5]
```diff
@@ -28,18 +28,25 @@ export class TelemetryUtils {
   }
 
   public static sendException(
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts:30]: I would suggest change function name to "sendErrorEvent"

File: packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts [hunk 6]
```diff
@@ -28,18 +28,25 @@ export class TelemetryUtils {
...
     }
-    properties[Constants.telemetryComponent] = Plugins.pluginNameComplex;
-    TelemetryUtils.ctx.telemetryReporter?.sendTelemetryException(
-      error,
+    
+    properties[Telemetry.component] = Plugins.pluginNameComplex;
+    properties[Telemetry.errorCode] = errorCode;
+    properties[Telemetry.errorType] = errorType;
+    properties[Telemetry.errorMessage] = errorMessage;
+    TelemetryUtils.ctx.telemetryReporter?.sendTelemetryEvent(
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts:46]: You should call telemetryReporter.sendTelemetryErrorEvent() API here.

File: packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts [hunk 7]
```diff
@@ -28,18 +28,25 @@ export class TelemetryUtils {
...
-  ) {
+  ): void {
     if (!properties) {
       properties = {};
     }
-    properties[Constants.telemetryComponent] = Plugins.pluginNameComplex;
-    TelemetryUtils.ctx.telemetryReporter?.sendTelemetryException(
-      error,
+    
+    properties[Telemetry.component] = Plugins.pluginNameComplex;
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts:42]: Need "success": "no" property.

File: packages/fx-core/src/plugins/resource/simpleauth/utils/telemetry.ts [hunk 8]
```diff
@@ -18,32 +18,29 @@ export class TelemetryUtils {
...
-        if (!properties) {
-            properties = {};
-        }
-        properties[Constants.Component] = Constants.SimpleAuthPlugin.id;
-        TelemetryUtils.ctx.telemetryReporter?.sendTelemetryErrorEvent(eventName, properties, measurements, errorProps);
-    }
-
     public static sendException(
-        error: Error,
+        eventName: string,
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/simpleauth/utils/telemetry.ts:26]: to sendErrorEvent()

File: packages/fx-core/src/plugins/resource/simpleauth/utils/telemetry.ts [hunk 9]
```diff
@@ -18,32 +18,29 @@ export class TelemetryUtils {
...
         if (!properties) {
             properties = {};
         }
-        properties[Constants.Component] = Constants.SimpleAuthPlugin.id;
-        TelemetryUtils.ctx.telemetryReporter?.sendTelemetryException(error, properties, measurements);
+        properties[Telemetry.component] = Constants.SimpleAuthPlugin.id;
+        properties[Telemetry.errorCode] = errorCode;
+        properties[Telemetry.errorType] = errorType;
+        properties[Telemetry.errorMessage] = errorMessage;
+        TelemetryUtils.ctx.telemetryReporter?.sendTelemetryEvent(
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/simpleauth/utils/telemetry.ts:40]: call sendTelemetryErrorEvent()

File: packages/fx-core/src/plugins/resource/aad/index.ts [hunk 10]
```diff
@@ -90,7 +90,7 @@ export class AadAppForTeamsPlugin implements Plugin {
         ctx.logProvider?.error(`Detailed error: ${e.innerError.message}`);
       }
       TelemetryUtils.init(ctx);
-      TelemetryUtils.sendException(stage, e.name, e instanceof UserError ? Telemetry.userError : Telemetry.systemError, e.message );
+      TelemetryUtils.sendErrorEvent(stage, e.name, e instanceof UserError ? Telemetry.userError : Telemetry.systemError, e.message );
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/aad/index.ts:93]: errorCode format should be ${e.source}.${e.name}

File: packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts (hunk 4 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts:18]: Fixed. Thanks!

File: packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts (hunk 5 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts]: Fixed. Thanks!

File: packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts (hunk 6 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts]: Fixed. Thanks!

File: packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts (hunk 7 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts:43]: Fixed. Thanks!

File: packages/fx-core/src/plugins/resource/simpleauth/utils/telemetry.ts (hunk 8 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/simpleauth/utils/telemetry.ts:27]: Fixed. Thanks!

File: packages/fx-core/src/plugins/resource/simpleauth/utils/telemetry.ts (hunk 9 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/simpleauth/utils/telemetry.ts]: Fixed. Thanks!

File: packages/fx-core/src/plugins/resource/aad/index.ts (hunk 10 above)
Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/index.ts:93]: Fixed. Thanks!

File: packages/fx-core/package.json (hunk 2 above)
Comment (KennethBWSong) [reply packages/fx-core/package.json]: Fixed. Thanks!

File: packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts [hunk 11]
```diff
@@ -15,31 +15,40 @@ export class TelemetryUtils {
...
       properties,
       measurements
     );
   }
 
-  public static sendException(
-    error: Error,
+  public static sendErrorEvent(
+    eventName: string,
+    errorCode: string,
```
Comment (chagong) [inline packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts:33]: It's kind of confusing the param name is errorCode but you actually pass error name value here, better rename it.

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/aad/utils/telemetry.ts]: Fixed. Thanks!

File: packages/fx-core/src/plugins/resource/simpleauth/utils/telemetry.ts [hunk 12]
```diff
@@ -18,32 +18,31 @@ export class TelemetryUtils {
...
-        properties?: { [key: string]: string },
-        measurements?: { [key: string]: number },
+        measurements?: { [key: string]: number }
     ) {
         if (!properties) {
             properties = {};
         }
-        properties[Constants.Component] = Constants.SimpleAuthPlugin.id;
-        TelemetryUtils.ctx.telemetryReporter?.sendTelemetryException(error, properties, measurements);
+        properties[Telemetry.isSuccess] = Telemetry.success;
```
Comment (adashen) [inline packages/fx-core/src/plugins/resource/simpleauth/utils/telemetry.ts:37]: success? in error

Comment (KennethBWSong) [reply packages/fx-core/src/plugins/resource/simpleauth/utils/telemetry.ts]: Fixed. Thanks!

**Review by adashen — APPROVED** [review]

**Review by chagong — APPROVED** [review]

**Pull request merged**